server/services/participants.js

```diff
@@ -148,6 +155,17 @@ const updateParticipant = async (participantInfo) => {
       },
       { history: participantInfo.history || [], userUpdatedAt: new Date().toJSON() }
     );
+    if (changes.interested === 'withdrawn') {
+      const cohort = await getAssignCohort({ participantId: participantInfo.id });
+      // ensure that a participant has a cohort before adding post hire status
+      if (cohort) {
+        await createPostHireStatus({
+          participantId: participantInfo.id,
+          status: postHireStatuses.failedCohort,
+          data: {},
```

Comment:
I would recommend store cohortId in data for future reference.

Reply:
I think it makes sense not to delete the entry, since we'll want that historical info to show on the participant and cohort details in future

Not sure about storing the cohortId, that seems to make sense to me to do - @dbayly-freshworks thoughts?

Reply:
If we delete the user from the cohort table, then a slot will open up. Space in cohorts is calculated from the cohorts total minus the number of users in that cohort. Saving the Cohort ID makes sense to me so we have a record of what happened. 

It seems like the system we have to keep track of cohort size isn't serving our needs. @fwkendall  Should we refactor it to add a `remaining allocation` column in the database instead of counting the number of remaining users.

Reply:
It sounds like what we have now will work, as long as we don't add the deletion

Do you think it's worth a refactor now? Or hold off for a bit and see if we need more flexibility?

Reply:
I'm on the fence to be honest. I think let's keep the function as is, but be ready to refactor if requirements change.